### ISSUE TITLE
Spotify looses connection after search

### ISSUE THREAD

**Issue by marcellendi** [clementine-player/Clementine#6147]
### Before posting

Please follow the steps below and check the boxes with [x] once you did the step.

- [ x] I checked the issue tracker for similar issues
- [x ] I checked the [changelog](https://github.com/clementine-player/Clementine/blob/master/Changelog) if the issue is already resolved
- [ x] I tried the latest Clementine build from [here](https://builds.clementine-player.org/)

### System information

Please provide information about your system and the version of Clementine used.

- Operating System: Ubuntu Studio 18.4 & window 7
- Clementine version: 1.3.1-485-g5e0caa512 (Ubuntu)

### Expected behaviour / actual behaviour
Recently ( a week?) the search on the Spotify plugin stops the connection with Spotify. Before the search I can play Spotify songs from my playlist (also top tracks are visible). After doing a search on Spotify all connection to Spotify is lost. I installed the stable version of Clementine at work (windows 7, different computer and network) and face the same problem. (

When starting Spotify from the terminal I get errors like " static void SpotifyClient::ConnectionErrorCallback(sp_session*, sp_error) Can not connect to Spotify "

I reported this issue in the Clementine Google Group and found out another user is facing the same problem using windows 10 and Debian 9. 

I contacted Spotify by chat and they say it has to be a fault on Clementine side.

### Steps to reproduce the problem (only for bugs)
* start Clementine 
* play Spotify song from playlist or Top Track (song should play)
* do a Search on Spotify (no result showing)
* if started Clementine from terminal you should see error coming
* play another Spotify song from playlist (song doesn't play)

**Comment by Kleyguerth** [comment]
Same issue here, Opensuse Tumbleweed. It has been happening for over a month for me. 

After searching the only way to play from spotify again is by restarting the application. Signing out and then signing in again makes it play for about one minute, then stops playing again.

**Comment by marcellendi** [comment]
I am not a developer, but can this issue be related to the depreciation of Libspotify? And if so will there be an alternative way to use Spotify in Clementine?

**Comment by justen** [comment]
@nwam have you tried playing a track without searching first? Behaviour I'm seeing on Ubuntu 18.04 / Clementine 1.3.1-333-gf854bc5 is:

* Start Clementine. Playing a Spotify track from a previously created playlist works as expected.
* Try searching. As soon as I search for anything, libspotify drops into the fault state you posted and gets stuck there. No spotify tracks can be played.

It looks like it's specifically the search call that is failing, but that's knocking over everything Spotify.

**Comment by nwam** [comment]
@justen I just tested and that is the case. No songs play after I try to search (which returns no results). I'm running Ubuntu 17.10 / Clementine 1.3.1.

**Comment by Fermman** [comment]
Same here Elementary OS 0.4.1 Loki, Clemetine Versión 1.3.1-399-g3134e09.
From Playlist works, but from search it chrashes

**Comment by gbrauen** [comment]
@nwam @justen I see similar behaviour. Searching always fails with the same 112 connection error but it does not necessarily kill spotify playback of tracks that are in a playlist. I think I was seeing that a couple of weeks back. 

Now playback seems pretty stable with tracks that are in my playlist. But there is no longer a way to get spotify tracks into a playlist. Albums disappeared years ago. Playlists disappeared last year. Search was until now the only option.

Ubuntu 18.04, Clementine 1.3.1-477-ga65dabc

**Comment by marcellendi** [comment]
Drag and drop from the Spotify app to Clementine is also no option due to a gstreamer error. Very frustrating. Seems to be the fault of the Spotifylib, which is not supported anymore by Spotify. Seems they slowly let it die.

**Comment by marcellendi** [comment]
Thanks @justen. It is a bit of work, but it works.

**Comment by gbrauen** [comment]
@justen Yes, thanks. I was not thinking of the 3 application solution when I said "no longer a way" to get a spotify track in. That will work - many problems can be solved with a good text editor.  :smile:

**Comment by lavacano201014** [comment]
Is there any sort of contact address for Spotify? I have half a mind to remind them that third party player support is the only reason I'm shelling out for premium, and if they don't want to support it anymore they're going to lose customers.

**Comment by fitojb** [comment]
> Is there any sort of contact address for Spotify?

Users’ only hope is their [“Ideas” platform](https://community.spotify.com/t5/Ideas/ct-p/newideas), but they won’t care about your report unless it gets 500 votes.

**Comment by kentsangkm** [comment]
This is my little script to add song from spotify to current playlist without using text editor

https://github.com/kentsangkm/ClementineSpotifyWorkaround/tree/master

**Comment by marcellendi** [comment]
Hi Kentsangkm, 

Thnx, this is the best workaround so far :)

Marcel

**Comment by ralgozino** [comment]
Same behaviour on Windows 10. @justen tip works on Windows also.

**Comment by gbrauen** [comment]
@kentsangkm Thanks, starting from your idea, I posted a Gist with a python script to create a xspf playlist of all the songs in a spotify album when given the album link and an output file name.
[https://gist.github.com/gbrauen/451f859046c05ccdadf355d645bcee02](https://gist.github.com/gbrauen/451f859046c05ccdadf355d645bcee02)

**Comment by kimmov-kipe** [comment]
Anyone to make (or modify?) a script to create a playlist from a Spotify playlist? (: Would that be a harder one to write? (I don't know anything about either playlist formats, sorry.)

With something like that we could just manage the playlist in Spotify client and run a script after adding/deleting music. And keep listening with a nicer interface, with all our local files, too.

**Comment by kentsangkm** [comment]
Hi all, I created another script to search song from spotify and import it

https://github.com/kentsangkm/clementine_spotify_search

@kimmov-kipe I think your request shouldn't be very hard. Please let me know if you still want to have it

**Comment by JulianVolodia** [comment]
Hi @kentsangkm how about refactoring Clementine? Could you try to integrate import feature from Spotify in Clementine?

**Comment by JulianVolodia** [comment]
@kentsangkm what you think? It looks like nice feature imho.

**Comment by kentsangkm** [comment]
@JulianVolodia yes, I can try.

**Comment by kimmov-kipe** [comment]
@kentsangkm That would be great, thank you! 
Even though, if it could be integrated to Clementine itself, that would be even more awesome of course.

**Comment by JulianVolodia** [comment]
@kentsangkm if you have any question or want some pair programming or feedback return to us.

**Comment by kentsangkm** [comment]
Hi all. a new search provider is added and merged to master. You can search song from Spotify with it
